apps/src/code-studio/components/progress/TeacherUnitOverview.tsx

```diff
@@ -277,34 +278,53 @@ const TeacherUnitOverview: React.FC<TeacherUnitOverviewProps> = props => {
 
   React.useEffect(() => {
     if (!unitName && selectedSection?.unitName) {
-      navigate(selectedSection.unitName, {replace: true});
+      navigate(
+        generatePath(
+          LABELED_TEACHER_NAVIGATION_PATHS.unitOverview.absoluteUrl,
+          {unitName: selectedSection.unitName, sectionId: selectedSection.id}
+        ),
+        {replace: true}
+      );
+      return;
     }
-  });
+  }, [unitName, selectedSection, navigate]);
 
   React.useEffect(() => {
     if (!unitName || !userType || !userId) {
       return;
     }
+
+    if (unitLoaded === unitName) {
```

Comment:
Do we have any concern about using `unitName` here instead of ID?  I am wondering if that could cause some issues with year over year units that have the same `name` but different `ids`.   Just a wondering...

Reply:
Name is a unique identifier (e.g. `csp3-2024`) that is used to determine which unit to load as well, so I don't think this is a concern.